src/main/java/io/debezium/connector/yugabytedb/YugabyteDBEventDispatcher.java

```diff
@@ -193,23 +199,41 @@ public void changeRecord(YBPartition partition,
                                  Object key, Struct value,
                                  OffsetContext offsetContext,
                                  ConnectHeaders headers) throws InterruptedException {
-            Objects.requireNonNull(value, "value must not be null");
+            if (!isNOOP) {
+                Objects.requireNonNull(value, "value must not be null");
+            } else {
+                LOGGER.info("Creating a NO_OP record");
+            }                       
 
             LOGGER.trace("Received change record for {} operation on key {}", operation, key);
 
             // Truncate events must have null key schema as they are sent to table topics without keys
-            Schema keySchema = (key == null && operation == Envelope.Operation.TRUNCATE) ? null
+            Schema keySchema = (key == null && (operation == Envelope.Operation.TRUNCATE || isNOOP)) ? null
                                  : dataCollectionSchema.keySchema();
             String topicName = topicSelector.topicNameFor((T) dataCollectionSchema.id());
 
-            SourceRecord record = new SourceRecord(partition.getSourcePartition(),
-              offsetContext.getOffset(),
-              topicName, null,
-              keySchema, key,
-              dataCollectionSchema.getEnvelopeSchema().schema(),
-              value,
-              null,
-              headers);
+            SourceRecord record;
+
+            if (!isNOOP) {
```

Comment:
nit: Whenever possible, try to have branches on true conditions instead of false. It avoids an extra hop in understanding the code -- more readable.

Basically, prefer
```java
if (isNOOP) {} else {}
```

instead of

```java
if (!isNOOP) {} else {}
```